Core/Src/main.c

```diff
@@ -87,16 +130,46 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
-
+  //Pull high in case device powered on with CS low
+  HAL_GPIO_WritePin( GPIOB, GPIO_PIN_8, GPIO_PIN_SET);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
   while (1)
   {
     /* USER CODE END WHILE */
-
+	  //Bring CS line to low to init communication
+	  HAL_GPIO_WritePin( GPIOB, GPIO_PIN_8, GPIO_PIN_RESET);
+
+	  hal_status = HAL_SPI_TransmitReceive(&hspi1, tx_data, rx_data,
+			  NUM_BITS_TO_SEND, SPI_TIMEOUT);
+
+	  if (hal_status == HAL_OK)
+	      {
+	          /*first byte doesn't matter i think (repeated question
+	           * marks in documentation) first 5 bits in second byte
+	           * don't matter. Sixth is null
+	           *  THEREFORE, we only need to set the last 2 bits of
+	           *  the second byte. ie, use 3 or 0x00000011,. the
+	           *  previously defined mask
+		  	  */
+			  second_byte = rx_data[1] & BIT_MASK;
+			  third_byte = rx_data[2];
+
+			  //Append all 10 bits together
+			  pot_response = second_byte | third_byte;
```

Comment:
So remember, second byte and third byte are both uint8_t so when you bitwise OR them together, it will result in a 8 bit value, but what you want is a 16 bit number, so what you have to do a bit shift (i won't tell you how much) on the second byte before ORing